### PR TITLE
Allowing arm64 macOS to debug dotnet projects

### PR DESCRIPTION
**Pull request by arthurlockman**
Issue #4277

This commit removes the bit of code that was preventing the .net debugger from starting on Apple Silicon.
I've added a warning message that indicates that you might see unexpected issues when running this way.
This should start working once macOS 11.1 is released next week. (See https://github.com/dotnet/runtime/issues/44958 for more details)

### CODE REVIEW

File: src/coreclr-debug/activate.ts
Comment (gregg-miskelly) [inline src/coreclr-debug/activate.ts:46]: This isn't enough - the debugger will still not download. You can likely address that by adding the following to package.json:

```jsonc
    {
      "id": "Debugger",
      "description": ".NET Core Debugger (macOS / x64)",
      "url": "https://download.visualstudio.microsoft.com/download/pr/6f481c2a-74a8-41cc-a115-f2d059242062/96e3d82b95ff1375347fedf0f5c7d4e7/coreclr-debug-osx-x64.zip",
      "fallbackUrl": "https://vsdebugger.blob.core.windows.net/coreclr-debug-1-23-8/coreclr-debug-osx-x64.zip",
      "installPath": ".debugger",
      "platforms": [
        "darwin"
      ],
      // Line 275 of package.json -- Add 'arm64'
      "architectures": [
        "x86_64",
        "arm64"
      ],
      "binaries": [
        "./vsdbg-ui",
        "./vsdbg"
      ],
      "installTestPath": "./.debugger/vsdbg-ui",
      "integrity": "6BAA9A20DAA71018B9912A3E2A48C52A273ADC2E70F7101C12F688F397AD16D8"
    },
```

**Review by gregg-miskelly — CHANGES_REQUESTED** [review]
You are definitely welcome to do this now if you like, but according to [this comment](https://github.com/dotnet/runtime/issues/44958#issuecomment-743091812) this scenario will not work. So it probably makes sense for that to be resolved before this is merged.

File: src/coreclr-debug/activate.ts
Comment (arthurlockman) [reply src/coreclr-debug/activate.ts:46]: Alright, fixed that.

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=4288) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=4288) before we can accept your contribution.<br/><br/>:x: arthurlockman [sign now](https://cla.dotnetfoundation.org/OmniSharp/omnisharp-vscode?pullRequest=4288)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OmniSharp/omnisharp-vscode?pullRequest=4288) it.</sub>

**Comment by arthurlockman** [comment]
> You are definitely welcome to do this now if you like, but according to [this comment](https://github.com/dotnet/runtime/issues/44958#issuecomment-743091812) this scenario will not work. So it probably makes sense for that to be resolved before this is merged.

Yeah, I've been following along with that thread. It looks like it'll let you debug and hit breakpoints, but it'll crash if you try to do a single step. So it seems like it's half-working.

**Comment by gregg-miskelly** [comment]
> Yeah, I've been following along with that thread. It looks like it'll let you debug and hit breakpoints, but it'll crash if you try to do a single step. So it seems like it's half-working.

**Comment by gregg-miskelly** [comment]
To make sure, with these changes were you able to confirm that with these changes the debugger is successfully download and you can at least use 'Start Without Debugging' on the target app?

**Comment by arthurlockman** [comment]
> To make sure, with these changes were you able to confirm that with these changes the debugger is successfully download and you can at least use 'Start Without Debugging' on the target app?

Could you point me to how I could get tests to run with the Apple Silicon build of VS code? It's currently only on the 1.53.0-exploration build. I'm not that familiar with how the test framework works for VS extensions.

**Comment by gregg-miskelly** [comment]
@arthurlockman I believe you can just test this manually. 

1. Follow [these instructions](https://github.com/omnisharp/omnisharp-vscode#development) to launch an "experiment instance" of VS Code using your locally built extension.
2. In the experimental instance, open a folder with a hello world C# project
3. The debugger should automatically download and your new warning should get printed.
4. You can then launch the project using "Start without debugging"

**Comment by arthurlockman** [comment]
@gregg-miskelly alright, got it working.

The debugger successfully installs with the warning I put in:
debugger-install

Then the debugger starts and crashes (as expected):
debugger-run

Run without debugging works successfully!
run-without-debug

**Review by gregg-miskelly — APPROVED** [review]

**Comment by gregg-miskelly** [comment]
Thanks for doing this! I will merge this once the CI completes.

File: package.json
Comment (lutzroeder) [inline package.json:277]: @arthurlockman @gregg-miskelly do you have some context why `-x64.zip` is installed for `arm64`?

Comment (gregg-miskelly) [reply package.json:277]: In order to debug x64 processes on an M1

**Pull request merged**
